mypy/plugin.py

```diff
@@ -379,3 +421,346 @@ def int_pow_callback(ctx: MethodContext) -> Type:
+
+    def __init__(self, name: str, type: Optional[Type],
+                 has_default: bool, init: bool,
+                 context: Context) -> None:
+        self.name = name
+        self.type = type
+        self.has_default = has_default
+        self.init = init
+        self.context = context
+
+    def argument(self) -> Argument:
+        """Return this attribute as an argument to __init__."""
+        # Convert type not set to Any.
+        _type = self.type or AnyType(TypeOfAny.unannotated)
+        # Attrs removes leading underscores when creating the __init__ arguments.
+        return Argument(Var(self.name.lstrip("_"), _type), _type,
+                        None,
+                        ARG_OPT if self.has_default else ARG_POS)
+
+
+def attr_class_maker_callback(
+        attr_classes: Dict[TypeInfo, List[Attribute]],
+        ctx: ClassDefContext,
+        auto_attribs_default: bool = False
+) -> None:
+    """Add necessary dunder methods to classes decorated with attr.s.
+
+    attrs is a package that lets you define classes without writing dull boilerplate code.
+
+    At a quick glance, the decorator searches the class body for assignments of `attr.ib`s (or
+    annotated variables if auto_attribs=True), then depending on how the decorator is called,
+    it will add an __init__ or all the __cmp__ methods.  For frozen=True it will turn the attrs
+    into properties.
+
+    See http://www.attrs.org/en/stable/how-does-it-work.html for information on how attrs works.
+    """
+
+    def get_callable_type(call: CallExpr) -> Optional[CallableType]:
+        """Get the CallableType that's attached to a CallExpr."""
+        callable_type = None
+        if (isinstance(call.callee, RefExpr)
+                and isinstance(call.callee.node, Var)
+                and call.callee.node.type):
+            callee_type = call.callee.node.type
+            if isinstance(callee_type, Overloaded):
+                # We take the last overload.
+                callable_type = callee_type.items()[-1]
+            elif isinstance(callee_type, CallableType):
+                callable_type = callee_type
+        return callable_type
+
+    def get_argument(call: CallExpr, name: str) -> Optional[Expression]:
+        """Return the expression for the specific argument."""
+        callee = get_callable_type(call)
+        if not callee:
+            return None
+
+        argument = callee.argument_by_name(name)
+        if not argument:
+            return None
+
+        if not argument.name and not argument.pos:
+            return None
+
+        for i, (attr_name, attr_value) in enumerate(zip(call.arg_names, call.args)):
+            if argument.pos is not None and not attr_name and i == argument.pos:
+                return attr_value
+            if attr_name == argument.name:
+                return attr_value
+        return None
+
+    def get_bool_argument(expr: Expression, name: str, default: bool) -> bool:
```

Comment:
Move this function to the outer scope and prefix it with `_attrs_` to indicate it is a private helper for `attrs` plugin.